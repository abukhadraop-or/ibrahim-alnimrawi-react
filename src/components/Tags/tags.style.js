import styled from "styled-components";
export const TagsDiv = styled.div`
  align-items: center;
  border-radius: 0.25rem;
  color: #373a3c;
  display: flex;
  height: 3rem;
  justify-content: center;
  position: relative;
  width: 8rem;
`;
export const Tag = styled.div`
  align-items: center;
  background-color: ${(props) => (props.selected ? `#2F3236` : `#7d858e`)};
  border-radius: 0.35rem;
  border: none;
  color: #fff;
  cursor: pointer;
  display: flex;
  justify-content: center;
  margin: 0.25rem 0.5rem;
  max-height: 2.5rem;
  min-width: 4rem;
  padding: 0.25rem;
`;

export const PopTags = styled.div`
  display: flex;
  flex-wrap: wrap;
`;
export const TagsContainer = styled.div`
  background-color: #f3f3f3;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: fit-content;
  margin-right: 3rem;
  max-width: 20rem;
  padding-bottom: 2rem;
  @media (max-width: 550px) {
    margin: 0 auto;
  }
`;
export const NotFound = styled.div`
  color: #373a3c;
  margin: 1rem 3rem;
`;
