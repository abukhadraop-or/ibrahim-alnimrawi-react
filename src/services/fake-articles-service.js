const articles = [
  {
    articles: [
      {
        slug: 'how-to-train-your-dragon-1',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-2',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-3',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-4',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-5',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-6',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-7',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-8',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-9',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-10',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['reactjs', 'dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-11',
        title:
          'How to train your dragon How to train your dragon How to train your dragon',
        description:
          'Ever wonder how? wfhdghkv whbdskvwdd hgkdvd  vkdwhsgvdhsvhds dughdsuoug w8dug8ud gds8u 8u8go8ods8u dsus8gvu sd',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 25,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-12',
        title: 'How to train your dragon 2',
        description: 'So toothless',
        body: 'It a dragon',
        tagList: ['dragons', 'training', 'reactjs'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-13',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training', 'reactjs'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-14',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-15',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-16',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-17',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-18',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-19',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-20',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-21',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-22',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-23',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-24',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-25',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-26',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-27',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-28',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-29',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-30',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-31',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-32',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-33',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: 'It takes a Jacobian',
        tagList: ['dragons', 'training'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
    ],
    articlesCount: 23,
  },
];

export const paginate = (data, pageNumber, pageSize) => {
  const start = (pageNumber - 1) * pageSize;
  const end = start + pageSize;
  return {
    articles: data[0].articles.slice(start, end),
    articlesCount: data[0].articlesCount,
  };
};

/**
 * Get slice of articles array.
 *
 * @param {String} tag        Tag name.
 * @param {Number} pageNumber Number of the current page.
 * @param {Number} pageSize   Number of articles per page.
 *
 * @return {Promise<Object[]>} Promise fulfilled with articles.
 */
export const getArticles = async (tag, pageNumber, pageSize) => {
  if (tag) {
    const filteredArticles = articles.map((article) =>
      article.articles.filter((tg) => tg.tagList.includes(tag))
    );
    const finalFilter = [
      {
        articles: filteredArticles[0],
        articlesCount: filteredArticles[0].length,
      },
    ];

    return Promise.resolve(paginate(finalFilter, pageNumber, pageSize));
  }

  const allArticles = paginate(articles, pageNumber, pageSize);
  return Promise.resolve(allArticles);
};
