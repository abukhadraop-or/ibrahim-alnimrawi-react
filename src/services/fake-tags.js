const tags = {
  tags: ['reactjs', 'angularjs'],
};
/**
 * Get array of popular tags.
 *
 * @return {Promise<String[]>} Promise fulfilled with tags.
 */
export const getTags = () => Promise.resolve(tags);
