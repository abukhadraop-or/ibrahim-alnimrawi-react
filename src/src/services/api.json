{
  "api": "https://api.realworld.io/api/"
}
