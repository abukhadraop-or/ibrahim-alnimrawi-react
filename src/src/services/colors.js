const colors = {
  moderateLimeGreen: "#5cb85c",
  darkCyan_limeGreen: "#09733a",
  white: "#ffffff",
  gray: "#bbbbbb",
  darkGray: "#999999",
  veryDarkGrayishBlue: "#373a3c",
  veryLightGray: "#dddddd",
  lightGrayishBlue: "#b8c3e0",
  veryDarkGrayishBlue2: "#404b59",
  black: "#000000",
  veryDarkGrayishBlue3: "#2F3236",
  darkGrayishBlue: "#7d858e",
  veryLightGray_mostlyWhite: "#f3f3f3",
};
export default colors;
