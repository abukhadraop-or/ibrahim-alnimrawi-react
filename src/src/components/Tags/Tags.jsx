import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";
import { getTags } from "../../services/fakeTags";
import { NotFound, PopTags, Tag, TagsContainer, TagsDiv } from "./tags.style";

/**
 * Render a <Input> component
 *
 * @param {Function} props.onTagChange Function to handle tag change.
 * @param {Boolean} props.unClick      Boolean to check if tag un clicked.
 *
 * @return {JSX.Element}
 */
const Tags = ({ onTagChange, unClick, selcet }) => {
  const [tags, setTags] = useState([]);
  const [select, setSelect] = useState("");
  useEffect(() => {
    const data = async () => {
      let response = await getTags();
      const result = response.tags;
      setTags(result);
    };

    data();
  }, []);

  return (
    <TagsContainer>
      <TagsDiv>Popular Tags</TagsDiv>
      {!tags && <NotFound> No tags are here...yet</NotFound>}
      <PopTags>
        {tags.map((tag) => (
          <Tag
            selected={select === tag && !unClick}
            key={tag}
            onClick={() => {
              onTagChange(tag);
              setSelect(tag);
            }}
          >
            {tag}
          </Tag>
        ))}
      </PopTags>
    </TagsContainer>
  );
};
export default Tags;

Tags.propTypes = {
  onTagChange: PropTypes.func,
  unClick: PropTypes.bool,
};
