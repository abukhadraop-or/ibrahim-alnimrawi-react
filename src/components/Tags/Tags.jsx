import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { getTags } from 'services/fake-tags';
import { NotFound, PopTags, Tag, TagsContainer, TagsDiv } from './tags.style';

/**
 * Render a <Input> component
 *
 * @param {Object}   props             The props object.
 * @param {Function} props.onTagChange Function to handle tag change.
 * @param {String}   props.selectedTag The selected tag.
 *
 * @return {JSX.Element}
 */
const Tags = ({ onTagChange, selectedTag }) => {
  const [allTags, setAllTags] = useState([]);
  // handle initial state.
  useEffect(() => {
    const data = async () => {
      const response = await getTags();
      const result = response.tags;
      setAllTags(result);
    };

    data();
  }, []);

  return (
    <TagsContainer>
      <TagsDiv>Popular Tags</TagsDiv>
      {!allTags.length && <NotFound> No tags are here...yet</NotFound>}
      <PopTags>
        {allTags.map((tag) => (
          <Tag
            selected={tag === selectedTag}
            key={tag}
            onClick={() => {
              onTagChange(tag);
            }}
          >
            {tag}
          </Tag>
        ))}
      </PopTags>
    </TagsContainer>
  );
};
export default Tags;

Tags.propTypes = {
  onTagChange: PropTypes.func,
  selectedTag: PropTypes.string,
};

Tags.defaultProps = {
  onTagChange: null,
  selectedTag: '',
};
