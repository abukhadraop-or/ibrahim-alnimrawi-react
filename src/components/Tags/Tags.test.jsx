import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { getTags } from 'services/fake-tags';
import { NotFound } from '../Articles/articles.style';
import Tags from './Tags';

jest.mock('services/fake-tags.js', () => ({
  getTags: jest.fn(),
}));

Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {
  it('should match the snapshot be', () => {
    const tree = renderer.create(<Tags />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be `No tags are here...yet`', () => {
    const result = { tags: ['reactjs'] };
    getTags.mockResolvedValueOnce(result);
    act(() => {
      const wrapper = mount(<Tags />);
      const notFound = wrapper.find(NotFound);
    });
  });
});
