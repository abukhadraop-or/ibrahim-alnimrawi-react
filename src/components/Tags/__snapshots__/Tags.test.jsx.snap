// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match the snapshot be 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  color: #373a3c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  background-color: #f3f3f3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin-right: 3rem;
  max-width: 20rem;
  padding-bottom: 2rem;
}

@media (max-width:34.375rem) {
  .c0 {
    margin: 0 auto;
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Popular Tags
  </div>
  <div
    className="c2"
  />
</div>
`;
